src/rebuild/srv.c

```diff
@@ -160,6 +160,23 @@ is_rebuild_global_done(struct rebuild_global_pool_tracker *rgt)
 
 }
 
+/* determine if "most" engines are done with their current rebuild phase (scan or pull) */
+static bool
+is_rebuild_phase_mostly_done(int engines_done_ct, int engines_total_ct)
+{
+	const int MIN_WAIT_CT = 1;
+	const int MAX_WAIT_CT = 20;
+	int       wait_ct_threshold;
+	int       engines_waiting_ct = engines_total_ct - engines_done_ct;
+
+	/* When is the global operation "mostly done" (e.g., waiting for 20 or fewer engines)? */
+	wait_ct_threshold = engines_done_ct - .95 * engines_total_ct;
+	wait_ct_threshold = max(MIN_WAIT_CT, wait_ct_threshold);
+	wait_ct_threshold = min(MAX_WAIT_CT, wait_ct_threshold);
+
```

Comment:
I am not sure if this could work well in the small system, eg engines_total_ct is 5, if there are 2 engines not finished, then it will never mark mostly done and skip warning messages bumping?

Reply:
For smaller scale systems, we could determine that MIN_WAIT_CT=2 instead of 1. That would allow us to warn when 2 engines are not done in some smaller scale systems.

Also, now that I look at this code again, I'm not sure about the first equation for wait_ct_threshold (line 173). Maybe that can more simply be wait_ct_threshold = .05 * engines_total_ct.

Making the above two changes, the transition from a threshold of waiting on 2 engines to waiting on 3 engines happens at 60 engines. Then, every additional 20 engines in the system would correspond to increasing the wait_ct_threshold by 1. And at 400 engines or higher, the wait_ct_threshold would always be 20.